pkg/handlers/launch.go

```diff
@@ -144,23 +144,119 @@ type launchHandler struct {
 	lastFailureTime      map[string]time.Time
 	lastFailureTimeMutex sync.Mutex
 
+	processToWaitFor       chan k6.TestRun
+	cancelWaitForProcesses context.CancelFunc
+	waitForProcessesDone   chan struct{}
+	ctx                    context.Context
```

Comment:
Unfortunately, I cannot completely get rid of struct-level context since that struct implements a http.Handler and I need the context outside of the request handler. I've limited its use now with https://github.com/grafana/flagger-k6-webhook/pull/153/commits/1decf4634aeda84e68af83d4ab0aebb4c272f7d8, though 🙂